packages/wallets/src/features/cashier/flows/WalletTransfer/__tests__/WalletTransfer.spec.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { useTransferBetweenAccounts } from '@deriv/api';
+import { render, screen } from '@testing-library/react';
+import WalletTransfer from '../WalletTransfer';
+
+jest.mock('../../../../../components', () => ({
+    ...jest.requireActual('../../../../../components'),
+    Loader: jest.fn(() => <div>Loading</div>),
+}));
+
+jest.mock('../../../modules', () => ({
+    ...jest.requireActual('../../../modules'),
+    TransferModule: jest.fn(() => <div>TransferModule</div>),
+}));
+
+jest.mock('../../../screens/TransferNotAvailable', () => ({
+    ...jest.requireActual('../../../screens/TransferNotAvailable'),
+    TransferNotAvailable: jest.fn(() => <div>TransferNotAvailable</div>),
+}));
+
+jest.mock('@deriv/api', () => ({
+    ...jest.requireActual('@deriv/api'),
+    useTransferBetweenAccounts: jest.fn(),
+}));
+
+const mockUseTransferBetweenAccounts = useTransferBetweenAccounts as jest.MockedFunction<
+    typeof useTransferBetweenAccounts
+>;
+
+describe('<WalletTransfer />', () => {
+    it('should show the loader if the API response has not yet arrived', () => {
+        // @ts-expect-error - since this is a mock, we only need partial properties of the hook
+        mockUseTransferBetweenAccounts.mockReturnValue({
+            isLoading: true,
+            mutate: jest.fn(),
+        });
+
+        render(<WalletTransfer />);
+        expect(screen.getByText('Loading'));
+    });
+
+    it('should render the TransferNotAvailable screen as the API response is received', () => {
+        mockUseTransferBetweenAccounts.mockReturnValue({
+            data: {
+                // @ts-expect-error - since this is a mock, we only need partial properties of the hook
+                accounts: {},
+            },
+            mutate: jest.fn(),
+        });
+
+        render(<WalletTransfer />);
+        expect(screen.getByText('TransferNotAvailable'));
+    });
+});
```

Comment:
Hi team,
Need your opinion on this. Is this the right way to test parts of a component where we encounter such nested components.

Reply:
IMO - its valid way, good when child component is a hassle to run / requires lots of data mocking / contains third parties etc.

Reply:
I haven't tested, but from what I see in the code, I could remove "TransferModule" without tests failing - and that shouldn't be possible,

Reply:
Generally speaking, purist will say that we should mock all the child components all the time, to have actually pure unit tests. As unit test should test specific component, and ideally, shoudln't test the child component (so if I refactor child component, only that child component test will be broken, and not all the tests of all the places which are using it). I haven't been a fan of this, but convinced myself its good idea while writing this comment, so yeah, mocking sounds good :D